lib/remote_persistent_term.ex

```diff
@@ -307,9 +320,35 @@ defmodule RemotePersistentTerm do
 
   defp download_and_store_term(state, deserialize_fun, put_fun) do
     with {:ok, term} <- state.fetcher_mod.download(state.fetcher_state),
-         {:ok, decompressed} <- maybe_decompress(state, term),
-         {:ok, deserialized} <- deserialize_fun.(decompressed) do
-      put_fun.(deserialized)
+         {:ok, decompressed} <- maybe_decompress(state, term) do
+      try_deserialize_and_store(state, decompressed, deserialize_fun, put_fun)
+    end
+  end
+
+  defp try_deserialize_and_store(state, term, deserialize_fun, put_fun) do
+    case deserialize_fun.(term) do
+      {:ok, deserialized} ->
+        put_fun.(deserialized)
+
+      {:error, _reason} = error when state.version_fallback? ->
+        Logger.error(
+          "#{state.name} - failed to deserialize remote term, falling back to previous version"
+        )
+
+        case state.fetcher_mod.previous_version(state.fetcher_state) do
```

Comment:
lets avoid this nested case by extracting it in to a new function, maybe `try_previous_version/3`?